tests/pytest/ui-tests/conftest.py

```
@@ -0,0 +1,161 @@
+            ['kubectl', 'port-forward', '-n', 'default', 'service/ark-dashboard', '3274:3000'],
+            stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        time.sleep(5)
+        
+        if port_forward.poll() is not None:
+            pytest.exit("Port forwarding failed", returncode=1)
+        
+        wait_for_dashboard()
+        
+        yield
+    finally:
+        if port_forward:
+            port_forward.terminate()
+            port_forward.wait(timeout=5)
+
+
+@pytest.fixture(scope="session")
+def playwright():
+    with sync_playwright() as p:
+        yield p
+
+
+@pytest.fixture(scope="session")
+def browser(playwright, ark_setup, request):
+    visible = request.config.getoption("--visible")
+    browser_type = request.config.getoption("--browser-type")
+    
+    if browser_type == "gecko":
+        browser_type = "firefox"
+    
+    launch_args = {
+        "headless": not visible,
+        "slow_mo": 500 if visible else 0,
+        "args": ["--start-maximized"] if visible else []
+    }
+    
+    browser = getattr(playwright, browser_type).launch(**launch_args)
+    yield browser
+    browser.close()
+
+
+@pytest.fixture(scope="session")
+def context(browser, request):
+    visible = request.config.getoption("--visible")
+    context_args = {
+        "viewport": None if visible else {"width": 1920, "height": 1080},
+        "ignore_https_errors": True
+    }
+    context = browser.new_context(**context_args)
+    yield context
+    context.close()
+
+
+@pytest.fixture(scope="session")
+def page(context):
+    page = context.new_page()
+    yield page
+    page.close()
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    outcome = yield
+    rep = outcome.get_result()
+    
+    if rep.when == "call" and rep.failed:
+        page = item.funcargs.get("page")
+        if page:
+            try:
+                page.screenshot(path=f"screenshots/{item.name}.png")
```

I dont see the `screenshots` folder, this could cause silent failures. Please create the directory where needed and add `.gitkeep` when empty